main.go

```diff
@@ -0,0 +1,213 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/scylladb/go-set/strset"
+	"github.com/spf13/pflag"
+	"go.uber.org/zap"
+)
+
+var (
+	validFormats = strset.New("png", "pdf")
+	skipDirs     = strset.New(".git")
+
+	// register the log-level flag natively to the stdlib flag package
+	logLevel = zap.LevelFlag("log-level", zap.InfoLevel, "What log level to use")
+)
+
+func main() {
+	// Setup logger
+	log := zap.NewExample().WithOptions(zap.IncreaseLevel(logLevel))
+	defer func() { _ = log.Sync() }()
+	zap.ReplaceGlobals(log)
+	if err := run(); err != nil {
+		log.Fatal(err.Error())
+	}
+}
+
+func run() error {
+	// Setup default config
+	cfg := &Config{
+		RootDir: "/files",
+		Formats: []string{"png", "pdf"},
+		SubDirs: []string{"."},
+		Files:   map[string]string{},
+	}
+
+	// Unconditionally overwrite root path if GH Action workspace is set
+	if ghWorkspace := os.Getenv("GITHUB_WORKSPACE"); len(ghWorkspace) != 0 {
```

Comment:
maybe we can just pass this param from the action, to avoid having this "hack" in the code?